Libraries/src/Amazon.Lambda.Annotations.SourceGenerators/LambdaFunctionCodeGenerator.cs

```diff
@@ -0,0 +1,261 @@
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using System.Linq;
+using System.Text;
+using System.Security.Cryptography;
+
+namespace Amazon.Lambda.Annotations.SourceGenerators
+{
+    public class LambdaFunctionCodeGenerator
+    {
+        private readonly MethodDeclarationSyntax _lambdaFunctionMethodSyntax;
+        private readonly ClassDeclarationSyntax _startupSyntax;
+        private readonly GeneratorExecutionContext _context;
+        private IMethodSymbol _lambdaFunctionMethodModel;
+        private IMethodSymbol _configureMethodSymbol;
+
+        /// <summary>
+        /// Returns class name of the generated method.
+        /// </summary>
+        public string ClassName => $"{LambdaFunctionMethodSymbol.ContainingType.Name}_{LambdaFunctionMethodSymbol.Name}_Generated";
+
+        /// <summary>
+        /// Represents simplified Lambda function,
+        /// It may or may not have the parameters such as customer input and ILambdaContext.
+        /// </summary>
+        public IMethodSymbol LambdaFunctionMethodSymbol
+        {
+            get
+            {
+                if (_lambdaFunctionMethodModel == null)
+                {
+                    var methodModel = _context.Compilation.GetSemanticModel(_lambdaFunctionMethodSyntax.SyntaxTree);
+                    _lambdaFunctionMethodModel = methodModel.GetDeclaredSymbol(_lambdaFunctionMethodSyntax);
+                }
+
+                return _lambdaFunctionMethodModel;
+            }
+        }
+
+        public INamedTypeSymbol ReturnType => LambdaFunctionMethodSymbol.ReturnType as INamedTypeSymbol;
+
+        /// <summary>
+        /// Represents the Configure(IServiceCollection) method in the LambdaStartup attributed class.
+        /// </summary>
+        /// <exception cref="NotSupportedException">Thrown when multiple LambdaStartup attributed classes exist in the project.</exception>
+        public IMethodSymbol ConfigureMethodSymbol
```

Comment:
Semantic code searching is very meta and will always be hard to understand. Please add lots of inline comments of exactly what you are attempting. This is a general rule for all generator code.